File: my-leetcode-solutions/reverse-linked-list.ts
// My solution to 206 - Reverse Linked List

/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function reverseList(head: ListNode | null): ListNode | null {
    
};

// Time Complexity:

class ListNode {
    val: number
    next: ListNode | null

    constructor(val?: number, next?: ListNode | null) {
        this.val = (val===undefined ? 0 : val)
        this.next = (next===undefined ? null : next)
    }
}

const headTestCases = [
    [1, 2, 3, 4, 5],
    [1, 2],
    []
];

for (let i = 0; i < headTestCases.length; i++) {
    let head: ListNode | null = null;

    for (let j = headTestCases[i].length - 1; j >= 0; j--) {
        head = new ListNode(headTestCases[i][j], head);
    }

    const output = reverseList(head);
    console.log(`Output: ${JSON.stringify(output)} for head as ${JSON.stringify(headTestCases[i])}`);
}

export {};

File: my-leetcode-solutions-py/reverse-linked-list.py
# My solution to 206 - Reverse Linked List

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: ListNode | None) -> ListNode | None:
        


# Time Complexity:

head_test_cases = [
    [1, 2, 3, 4, 5],
    [1, 2],
    [],
]

solution = Solution()

for i in range(len(head_test_cases)):
    head = None
    for value in reversed(head_test_cases[i]):
        head = ListNode(value, head)

    output = solution.reverseList(head)
    print(f"Output: {output!r} for head as {head_test_cases[i]!r}")
